FEATURE/ElvaRivle/BaseModal

ElvaRivle opened the pull request:
ElvaRivle/Feature/BaseModal (DON'T MERGE) 

--------------------------------------------------------------------
 MAIN
 * refactored folder structure and data flow 
 * resolved PR comments
--------------------------------------------------------------------
 OTHER
 * /
--------------------------------------------------------------------
 TODO
 * TODO in AddEmployeeModal.js

ElvaRivle committed:
FEATURE/ElvaRivle/BaseModal

--------------------------------------------------------------------
 MAIN
 * modal created
--------------------------------------------------------------------
 OTHER
 * /
--------------------------------------------------------------------
 TODO
 * populate modal from props
 * FIXME's left to fix

ElvaRivle committed:
FEATURE/ElvaRivle/BaseModal

--------------------------------------------------------------------
 MAIN
 * moved modal from common to modal folder
--------------------------------------------------------------------
 OTHER
 * /
--------------------------------------------------------------------
 TODO
 * populate modal from props
 * FIXME's left to fix

ElvaRivle committed:
FEATURE/ElvaRivle/BaseModal

--------------------------------------------------------------------
 MAIN
 * modalItem component added
 * populated modal from props
 * basic styling of modal
--------------------------------------------------------------------
 OTHER
 * modal example in Dashboard
--------------------------------------------------------------------
 TODO
 * FIXME's left to fix

ElvaRivle commented:
FEATURE/ElvaRivle/BaseModal

--------------------------------------------------------------------
 MAIN
 * modalItem component added
 * populated modal from props
 * basic styling of modal
--------------------------------------------------------------------
 OTHER
 * modal example in Dashboard
--------------------------------------------------------------------
 TODO
 * FIXME's left to fix

becarselman commented on src/components/dashboard/Dashboard.js:
```diff
@@ -1,10 +1,43 @@
+
   return (
     <>
+      <button onClick={() => setIsOpen(true)}>
```
move this onClick function outside of jsx. In this case function will be defined on each render.

becarselman commented on src/components/dashboard/Dashboard.js:
```diff
@@ -1,10 +1,43 @@
+        Click to Open Modal
+      </button>
+
+      <Modal handleClose={() => setIsOpen(false)} isOpen={isOpen} items = { employees } />
```
same for handleClose

becarselman commented on src/components/modal/modal.scss:
```diff
@@ -0,0 +1,29 @@
+    //width: 70%;
+    //height: 70%;
+    background-color: #E6F2ED;
+    color: #000;
```
move colors in variables file for colors. and if already exist reuse it

becarselman commented on src/components/modal/modal.scss:
```diff
@@ -0,0 +1,29 @@
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    //font-size: 2rem;
```
move commented code

becarselman commented on src/components/modal/modalItem.js:
```diff
@@ -0,0 +1,34 @@
+import "./modalItem.scss"
+
+const ModalItem = (props) => {
```
you can use useMemo for this type of component

ElvaRivle committed:
FEATURE/ElvaRivle/BaseModal

--------------------------------------------------------------------
 MAIN
 * fixed issues from PR review
--------------------------------------------------------------------
 OTHER
 * created [wiki](https://github.com/becarselman/ant-colony-pharaoh-frontend/wiki/Modal-Usage) for usage of modal
--------------------------------------------------------------------
 TODO
 * modal styling

ElvaRivle commented:
FEATURE/ElvaRivle/BaseModal

--------------------------------------------------------------------
 MAIN
 * fixed issues from PR review
--------------------------------------------------------------------
 OTHER
 * created [wiki](https://github.com/becarselman/ant-colony-pharaoh-frontend/wiki/Modal-Usage) for usage of modal
--------------------------------------------------------------------
 TODO
 * modal styling

ElvaRivle committed:
FEATURE/ElvaRivle/BaseModal

--------------------------------------------------------------------
 MAIN
 * created testing modal for creating employee/user
 * nicer styling of modal
 * added temporary route /employees to test modal
--------------------------------------------------------------------
 OTHER
 * renamed component files to Capital
 * put Departments and Stacks in separate file
 * added react-select package
--------------------------------------------------------------------

ElvaRivle committed:
FEATURE/ElvaRivle/BaseModal

--------------------------------------------------------------------
 MAIN
 * added password field
 * actually sending request to backend
--------------------------------------------------------------------
 OTHER
 * /
--------------------------------------------------------------------
 TODO
 * /

ElvaRivle committed:
baby steps towards rework

ElvaRivle committed:
everything renders

ElvaRivle committed:
it's functional !!

ElvaRivle committed:
revert old api_url, although wrong

ElvaRivle committed:
ElvaRivle/Feature/BaseModal

--------------------------------------------------------------------
 MAIN
 * sending user data with redux
 * refactored folder structure for (temporary)AddEmployeeModal
--------------------------------------------------------------------
 OTHER
 * /
--------------------------------------------------------------------
 TODO
 * refactor modal also
 * fix bugs and implement functionalities in sagas

becarselman commented on src/components/_addEmployeeModal/AddEmployeeModal.js:
```diff
@@ -0,0 +1,79 @@
+
+  return (
+    <>
+      <Modal handleClose={() => {
```
you can create separate function handleClose and just pass here. avoid using inline function in jsx

becarselman commented on src/components/modal/Modal.js:
```diff
@@ -0,0 +1,43 @@
+        <CloseOutlined onClick={handleClose} className="close-btn"/>
+        <div className="modal-content">
+          { mappedItems }
+          {
```
move this outside of jsx. It will be more readable

becarselman commented on src/components/modal/ModalItem.js:
```diff
@@ -0,0 +1,60 @@
+import "./modalItem.scss"
+import Select from "react-select";
+
+const ModalItem = (props) => {
```
this is good but my personal preference is to destruct items from props and than I don't need to type props each time. But this is also ok

ElvaRivle replied:
this component shall be deleted when refactoring starts, but I'll keep that in mind

becarselman commented on src/components/modal/modalItem.scss:
```diff
@@ -0,0 +1,16 @@
+  display: flex;
+  flex-direction: row;
+  justify-content: flex-start;
+  //line-height: 26px;
```
remove commented code if you don't need

becarselman reviewed (commented):
This looks good just fix this small comments. Also I hope you are tested everything create/ edit

ElvaRivle committed:
ElvaRivle/Feature/BaseModal

--------------------------------------------------------------------
 MAIN
 * refactored folder structure and data flow
 * resolved PR comments
--------------------------------------------------------------------
 OTHER
 * /
--------------------------------------------------------------------
 TODO
 * TODO in AddEmployeeModal.js

ElvaRivle committed:
Merge branch 'develop' into ElvaRivle/Feature/BaseModal
